### ISSUE TITLE
Damage bug against Umbral Mithrix mod

### ISSUE THREAD

**Issue by Arch-Noize** [Moffein/SniperClassic#24]
Hi! This is a bit of repost, technically, but I reported [this issue](https://github.com/FocusedFault/UmbralMithrix/issues/1) over the other mod in question. You'll find the info on the linked issue, but I'll send it over here too just in case. I don't know if its a compatibility issue or some behaviour that Umbral Mithrix has per se, I just know it happens with him specifically. I've tried replicating it with other enemies to no avail. 

> Hi! I uh... have no clue if this place will be of help to actually report a bug. I tried looking all around both the Thunderstore and any other place regarding ROR2 Modding buuuut I couldn't find anything so.
Anyways, I found an interesting bug I'm trying to replicate. In short, I'm playing the [Sniper Mod](https://thunderstore.io/package/EnforcerGang/SniperClassic/1.5.0/) character and he has a 'marker' ability. After a cooldown, hitting a weakspot, you deal extra damage to marked enemy and causes an explosion around them that repeats the damage to nearby enemies. Thing is, if you mark Mithrix and die.... the mark is both reset and kept. If you mark him again, you get seemingly infinite damage loop as the explosion retriggers itself infinite times with a short delay between each explosion, despite the spotter still going on cooldown.
The bug occurs as follows:
> - Have Mithrix marked, any form of Dio's Best Friend, and a Plasma Shrimp.
> - Die while fighting Mithrix, promptly get revived.
> - Mark Mithrix again, wait for weak spot to appear, trigger explosion
>- What proceeds should be an endlessly repeating explosion as if it were some crazy procc chain.
> I have no clue what triggers it and I'm trying to get a recording of it happening but OBS is being weird with me as of the time I'm writing this. Hope this helps or makes sense in any way whatsoever??? It's super weird and I don't know if its an issue with the modded survivor or some behavior Umbral Mithrix has?

**Comment by Moffein** [comment]
Next time you see the issue occur (or if you haven't booted the game up since the time the issue occurred), send your log file and I'll take a look at it.
You can find your log file by going to r2modman -> Settings -> Locations -> Browse Profile Folder, then going into the BepInEx folder and posting LogOutput.log here.

**Comment by Arch-Noize** [comment]
[LogOutput.log](https://github.com/Moffein/SniperClassic/files/12768428/LogOutput.log)
Alright so. I haven't been able to record it (I'm playing on Linux and OBS just... _dies_ when I try to start recording at the same time I have the game running, for some reason). BUT I've found to replicate it.

I did a Monsoon run with Aerolt to speed up the process of the run and give me the necessary items to try the error out. I only had to cheat in Plasma Shrimp and I.C.B.M, used noclip a couple times to move faster and just pass the run. Other than that, this run is completely normal / unaltered.

Basically, I've seen ATG consumes the mark (for whatever reason). Both ATG and Plasma Shrimp do this. However, if you die while Mithrix is marked and mark him again... your spotter doesn't go to cooldown. It just continues to be there without consumption. So, what I _think_ happens is:
- Mithrix is perma marked after you have him already marked and die, then mark again.
- You shoot him, proccing the mark and Plasma Shrimp.
- Plasma Shrimp proccs the mark, and somehow the instance of damage of the mark proccs Plasma Shrimp again
- Leading to an infinite loop until it dies

In the log output I only managed to make it 'procc' several times throughout the fight. Both with my own Dio's and the ones provided by Practice Mode from the mod. I have no clue how it looks in the log, or if there's any other way I can replicate it, but let me know how I can help.
